web/cat/mad_hatter/decorators.py

```diff
@@ -7,11 +7,20 @@
 
 
 log(tool)
+hooks = []
 
+def hook(priority=1, return_direct=False):
+    def cat_hook_wrapper(func):
+        def hook_func(*args, **kwargs):
+            # log(func)
+            return func(*args, **kwargs)
 
-def hook(func):
-    def cat_hook_wrapper(*args, **kwargs):
-        # log(func)
-        return func(*args, **kwargs)
-
+        hooks.append({
```

Comment:
We can write the hook decorator as a class and having this information as instance attributes.
So in the plugins hooks are simply functions, but we are more free on the hook decorator level.
What do you think?